src/table/data-grid-interaction-model.class.ts

```diff
@@ -0,0 +1,263 @@
+this.interactionModel.position.subscribe(event => {
+	const [currentRow, currentColumn] = event.current;
+	const [previousRow, previousColumn] = event.previous;
+
+	// query the TableAdapter for the cell at the current row and column ...
+	const currentElement = tableAdapter.getCell(currentRow, currentColumn);
+	// ... and make it focusable it
+	Table.setTabIndex(currentElement, 0);
+
+	// if the model has just initialized don't focus or reset anything
+	if (previousRow === -1 || previousColumn === -1) { return; }
+
+	// query the TableAdapter for the cell at the previous row and column ...
+	const previousElement = tableAdapter.getCell(previousRow, previousColumn);
+	// ... and make it unfocusable (now there is only a single focusable cell)
+	Table.setTabIndex(previousElement, -1);
+
+	// finally, focus the current cell (skipped during initilzation)
+	Table.focus(currentElement);
+});
+ * ```
+ */
+export class DataGridInteractionModel {
+	/**
+	 * An Observable that provides an aggregated view of the `rowIndex` and `columnIndex` Observables
+	 */
+	readonly position: Observable<DataGridPosition>;
+	/**
+	 * An Observable that provides the current and previous row indexes.
+	 */
+	readonly rowIndex: Observable<{ current: number, previous: number }>;
+	/**
+	 * An Observable that provides the current and previous column indexes.
+	 */
+	readonly columnIndex: Observable<{ current: number, previous: number }>;
+
+	/**
+	 * Internal subject to handle changes in row
+	 */
+	protected rowSubject = new BehaviorSubject({ current: 0, previous: -1 });
+	/**
+	 * Internal subject to handle changes in column
+	 */
+	protected columnSubject = new BehaviorSubject({ current: 0, previous: -1 });
+
+	/**
+	 * The latest value emitted by the rowSubject
+	 */
+	protected get currentRow() {
+		return this.rowSubject.getValue().current;
+	}
+
+	/**
+	 * The latest value emitted by the columnSubject
+	 */
+	protected get currentColumn() {
+		return this.columnSubject.getValue().current;
+	}
+
+	/**
+	 * The last column as reported by the adapter
+	 */
+	protected get lastColumn() {
+		return this.tableAdapter.lastColumn;
+	}
+
+	/**
+	 * The last row as reported by the adapter
+	 */
+	protected get lastRow() {
+		return this.tableAdapter.lastRow;
+	}
+
+	/**
+	 * `DataGridInteractionModel` requires knowledge of events, and a representation of your table/grid to be useful.
+	 *
+	 * @param keyboardEventStream an Observable of KeyboardEvents. Should be scoped to the table container.
+	 * @param clickEventStream an Observable of ClickEvents. should only include clicks that take action on items known by the TableAdapter
+	 * @param tableAdapter an instance of a concrete class that implements TableAdapter. The standard carbon table uses TableDomAdapter
+	 */
+	constructor(
+		protected keyboardEventStream: Observable<KeyboardEvent>,
+		protected clickEventStream: Observable<MouseEvent>,
+		protected tableAdapter: TableAdapter) {
```

Comment:
Would it make sense to move the `) {` to a new line in cases like this? Like
```suggestion
		protected tableAdapter: TableAdapter
	) {
```

Reply:
I don't think we have a case like this anywhere else! So, probably 🤔